lib/Reader.php

```diff
@@ -322,4 +328,35 @@ function getDeserializerForElementName($name) {
 
     }
 
+    /**
+     * Set the URI containing the XML to parse
+     * @param  string  $uri
+     * @param  string  $encoding
+     * @param  int $options
+     * @return bool            Returns TRUE on success or FALSE on failure
+     */
+    function open($uri, $encoding = null, $options = 0) {
+        return $this->reading = parent::open($uri, $encoding, $options);
+    }
+
+    /**
+     * Set the data containing the XML to parse
+     * @param  string  $source
+     * @param  string  $encoding
+     * @param  int $options
+     * @return bool            Returns TRUE on success or FALSE on failure
+     */
+    function xml($source, $encoding = null, $options = 0) {
+        return $this->reading = parent::xml($source, $encoding, $options);
+    }
+
+    /**
+     * Close the XMLReader input
+     * @return bool             Returns TRUE on success or FALSE on failure.
+     */
+    function close() {
```

Comment:
I'm curious if there's a need for override `close`. Could you just add a separate `->stop` function instead? The tie in with `close` seems kind of unimportant, but I could be wrong.